### PR TITLE
refactor: use keyof typeof to reduce duplication

### PR DESCRIPTION
**Pull request by domoritz**
Merge after #6355
This pull request should reduce code duplication. I also replaced some strings with references to variables. More could be done but this is a start.

### CODE REVIEW

**Comment by kanitw** [comment]
> Merge after #6357



#6357 is outdated and closed

**Comment by domoritz** [comment]
Sorry, the right one is #6355

File: src/channel.ts [hunk 1]
```diff
@@ -66,32 +66,36 @@ export const HREF: 'href' = 'href';
 
 export const URL: 'url' = 'url';
 
-export type PositionChannel = 'x' | 'y' | 'x2' | 'y2';
-
-const POSITION_CHANNEL_INDEX: Flag<PositionChannel> = {
+const POSITION_CHANNEL_INDEX = {
```
Comment (kanitw) [inline src/channel.ts:69]: FWIW, you declared 

```
const PositionScaleChannel = {
  x: 1,
  y: 1
} as const;
```

below, so perhaps we should be consistent.

File: src/compile/axis/properties.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import {Align, AxisOrient, Orient, ScaleType, SignalRef} from 'vega';
 import {isArray} from 'vega-util';
 import {Axis} from '../../axis';
 import {isBinned, isBinning} from '../../bin';
-import {PositionScaleChannel, X} from '../../channel';
+import {PositionScaleChannel, X, X2, Y, Y2} from '../../channel';
```
Comment (kanitw) [inline src/compile/axis/properties.ts:5]: Please either revert this file or rebase it to https://github.com/vega/vega-lite/pull/6312.

File: src/compile/facet.ts [hunk 3]
```diff
@@ -245,7 +245,7 @@ export class FacetModel extends ModelWithField {
         as.push(`distinct_${field}`);
       }
     } else {
-      for (const channel of ['x', 'y'] as ScaleChannel[]) {
+      for (const channel of [X, Y] as ScaleChannel[]) {
```
Comment (kanitw) [inline src/compile/facet.ts:248]: POSITION_SCALE_CHANNNELS?

File: src/compile/axis/properties.ts (hunk 2 above)
Comment (kanitw) [reply src/compile/axis/properties.ts:5]: actually https://github.com/vega/vega-lite/pull/6372

File: src/channel.ts (hunk 1 above)
Comment (domoritz) [reply src/channel.ts:69]: Let me fix the one below. I went back and forth between the two styles.

File: src/compile/axis/properties.ts (hunk 2 above)
Comment (domoritz) [reply src/compile/axis/properties.ts:5]: Hopefully it's easy enough to resolve the conflicts here. I am happy to do the rebase if needed.

File: src/compile/facet.ts (hunk 3 above)
Comment (domoritz) [reply src/compile/facet.ts]: Done. I also replaced a few more places I found in the code.

**Comment by domoritz** [comment]
I addressed your comments @kanitw

**Pull request merged**